src/app/api/core/utils/withErrorHandler.ts

```diff
@@ -39,8 +39,8 @@ export const withErrorHandler = (handler: RequestHandler): RequestHandler => {
       console.error(formattedError)
 
       // Default staus and message for JSON error response
-      let status: number = httpStatus.BAD_REQUEST
-      let message: string | ZodFormattedError<string> = 'Something went wrong'
+      let status: number = (error as StatusableError).status || httpStatus.BAD_REQUEST
+      let message: string | ZodFormattedError<string> = (error as MessagableError).body?.message || 'Something went wrong'
```

Comment:
The code uses type assertions like `error as StatusableError` and `error as MessagableError` repeatedly, which is risky if error doesn't match the expected structure. Instead of type casting, using type guards (custom functions to check if error is of a particular type) would be safer.

Reply:
Type guards aren't necessary here since we have a `||` clause following closesly by. If error doesn't satisfy `StatusableError` or `MessagableError`, the default value will simply substitute this.